### ISSUE TITLE
Make blocklist/safelist name argument optional

### ISSUE THREAD

**Issue by grzuy** [kickstarter/rack-attack#322]
```ruby
Rack::Attack.blocklist("this is the name of the blocklist") do |request|
  # ...
end
```

So far it seems that names, from the user perspective, are only potentially useful for when subscribing to notifications in order to distinguish between different blocks, safelists and throttles.

If this is the case, it seems like it would be better if users wouldn't be forced to pass the argument, while continuing to provide the feature by making it optional.

**Comment by lmansur** [comment]
Agreed. In this case, would we provide a default name?

**Comment by grzuy** [comment]
> Agreed.

Cool, thanks for the feedback

> In this case, would we provide a default name?

Not sure to be honest. Leaning towards not providing default unless there's a clear benefit to the user I think.

**Comment by grzuy** [comment]
`#throttle` uses the name as part of the cache key to count requests, so leaving it out of the scope of this issue for now.

**Comment by grzuy** [comment]
This is now available in `v6.0.0`.
